mongo/collection.go

```diff
@@ -1818,7 +1818,8 @@ func (coll *Collection) drop(ctx context.Context) error {
 	sess := sessionFromContext(ctx)
 	if sess == nil && coll.client.sessionPool != nil {
 		var err error
-		sess, err = session.NewClientSession(coll.client.sessionPool, coll.client.id, session.Implicit)
+		sess, err = session.NewClientSessionSetImplicit(coll.client.sessionPool, coll.client.id,
+			session.Implicit)
```

Comment:
I don't think we want to do this "implicit session only after connection" thing for dropping collections. It causes some very strange behavior such as

```
InsertMany error for collection test: bulk write exception: write errors: [E11000 duplicate key error collection: transaction-tests.test index: _id_ dup key: { _id: 0 }]
```